### PR TITLE
Replace cbor with bincode

### PR DESCRIPTION
**Pull request by madadam**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/maidsafe_utilities/32)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @ustulation (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: src/serialisation.rs [hunk 1]
```diff
@@ -15,45 +15,59 @@
...
 pub fn serialise<T>(data: &T) -> Result<Vec<u8>, SerialisationError>
     where T: Encodable
 {
-    let mut encoder = Encoder::from_memory();
-    try!(encoder.encode(&[data]));
-    Ok(encoder.into_bytes())
+    encode(data, SizeLimit::Infinite).map_err(From::from)
+}
+
+/// Serialise directly to a Writ
```
Comment (dirvine) [inline src/serialisation.rs]: Typo

**Comment by dirvine** [comment]
Nice, good to wrap the error type. We can probably remove all cbor references in all crates and implement From SerialisationError I think ?

File: src/serialisation.rs [hunk 2]
```diff
@@ -15,45 +15,44 @@
...
-    /// Result - they return an Option too.
+    /// Serialisation error.
     #[derive(Debug)]
     pub enum SerialisationError {
-        /// Encapsulated Cbor Error
-        CborError(err: ::cbor::CborError) {
-            description("Cbor error")
-            display("Cbor error: {}", err)
+        /// Error during serialisation (encoding).
+        SerialiseError(err: EncodingError) {
```
Comment (Fraser999) [inline src/serialisation.rs]: If we're changing these anyway, could we remove the `Error` suffix to keep Clippy happy?

Comment (madadam) [inline src/serialisation.rs]: Didn't know Clippy doesn't like that, but sure, why not. I'm anticipating more changes to this module anyway, as I'm porting everything to bincode. So don't merge yet.

**Comment by Fraser999** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_utilities/32)_

**Pull request merged**